pkg/segment/writer/segwriter.go

```diff
@@ -1042,29 +1044,76 @@ func BackFillPQSSegmetaEntry(segsetkey string, newpqid string) {
 			continue
 		}
 	}
-	wfd, err := os.OpenFile(localSegmetaFname, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0644)
+	err = WriteSegMeta(localSegmetaFname, preservedSmEntries)
 	if err != nil {
-		log.Errorf("BackFillPQSSegmetaEntry: Failed to open SegMetaFile name=%v, err:%v", localSegmetaFname, err)
-		return
+		log.Errorf("BackFillPQSSegmetaEntry: failed to write Segmeta: err=%v", err)
+	}
+}
+
+func WriteSegMeta(segmetaFname string, segMetaEntries []*structs.SegMeta) error {
+	wfd, err := os.OpenFile(segmetaFname, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0644)
```

Comment:
nit: maybe rename this to something like `WriteOverSegMeta` or `ClearAndWriteSegMeta` or similar, just to clarify that the old segmeta file gets removed rather than appended to

Reply:
addressed in the latest commit.